src/main/java/executor/service/handler/ProxyHandler.java

```diff
@@ -0,0 +1,7 @@
+package executor.service.handler;
+
+import executor.service.model.ProxyConfigHolder;
+
+public interface ProxyHandler {
```

Comment:
Here, I would suggest using Generic to unify the solution with ProxySourceClient, pull-request [#43](https://github.com/GFLCourses5/worker/pull/43).

![image](https://github.com/GFLCourses5/worker/assets/140727960/0d76556f-c3d4-4040-9b1d-a5acc8f74d66)

Using the generic ItemHandler<T> interface provides several benefits:
1. Code reuse: Generic interfaces allow you to create generic classes and methods that can operate on different data types without requiring code duplication. This promotes code reuse and reduces code complexity.
2. Type safety: Using generic interfaces provides type safety, which means that the compiler will control the types of data passed to the interface's methods. This helps prevent compile-time typing errors and improve code reliability.
3. Flexibility: Generic interfaces allow code to adapt to different data types, making the code more flexible and extensible. You can easily create implementations of the ItemHandler interface for different data types without having to create different interfaces for each type.

Reply:
1.As I said before **we dont need** magic instances that can do all type of work for all possible cases now.  In case when you'll change some behavior in one place you will have to change in another places.
2. What's wrong with type safety in my interface?
3.  Again look at p.1. Here you lose the semantic clarity that a custom functional interface like ProxyListener provides. Using a custom functional interface allows you to give a meaningful name to the callback method, making your code more self-explanatory and maintainable.